packages/core/src/core.ts

```diff
@@ -49,7 +50,23 @@ export class Colocated<T, L extends Location> {
   }
   protected value: T;
   protected key: symbol;
-  protected phantom?: (x: L) => void;
+}
+
+// Exported `Colocated` class and the corresponding covariant or contravariant type variations
+export class Colocated<T, L extends Location> extends ColocatedBase<T> {
+  phantom?: ((x: L) => void) & L;
```

Comment:
Question: do we need this line at all? It seems that everything typechecks without this line.